### DISCUSSION TITLE
Can Pseudobinary encode negative values?

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#555] [Q&A] [answered]
We're running into a situation where the stage may drop below zero. Does the pseudobinary format allow encoding and transmission of negative values?

**Comment by MikeNeilson** [comment, marked as answer]
Yeah, https://opendcs-env.readthedocs.io/en/latest/start-decoding.html#field-sensor-pseudo-binary, the docs aren't totally filled in yet but for decoding you use `I` instead of `B`

for encoding you'd have to check the manufacture, in doing it yourself it's still just standard two's compliment; so that "highest bit" based on your pseudo binary size, needs to be 1.

Not sure how much I've actually seen it in the wild so if there are still issues post the data here so we can help test. The mailing list would also be good, there's people that may not see these discussions there.
